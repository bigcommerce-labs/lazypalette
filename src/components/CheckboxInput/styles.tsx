import styled from 'styled-components';

export const Container = styled.div`
    position: relative;
`;

export const Input = styled.input.attrs({
    type: 'checkbox',
})`
    ${({ theme }) => `
        opacity: 0;

        :not(:checked) + label:after {
            opacity: 0;
        }

        :not(:checked) + label {
            background: #FFF;
            border: 1px solid ${theme.colors.guideText};
            box-sizing: border-box;
        }
    `}
`;

export const Label = styled.label`
    ${({ theme }) => `
        cursor: pointer;
        position: absolute;
        width: 1.25rem;
        height: 1.25rem;
        top: 25%;
        left: 0;
        background: ${theme.colors.primary};
        border-radius: 2px;

        :after {
            content: '';
            position: absolute;
            width: .5rem;
            height: .25rem;
            background: transparent;
            top: .25rem;
            left: .25rem;
            border: 3px solid ${theme.colors.stroke};
            border-top: none;
            border-right: none;
            transform: rotate(-45deg);
        }
    `}
`;

export const HiddenLabel = styled.label`
    padding-left: .5rem;
    line-height: 1.25rem;
    display: inline-block;
`;
