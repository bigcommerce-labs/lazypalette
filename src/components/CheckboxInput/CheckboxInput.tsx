import React, { PureComponent, ChangeEvent } from 'react';
import uuid from 'uuid';

import { Input, Label, Container, HiddenLabel } from './styles';
import { ThemeConfigChange } from '../../actions/theme';

interface CheckboxInputProps {
    inputId?: string;
    label?: string;
    name: string;
    checked: boolean;
    onChange?(configChange: ThemeConfigChange): void;
}

interface CheckboxInputState {
  checked: boolean;
  inputId: string;
}

class CheckboxInput extends PureComponent<CheckboxInputProps, CheckboxInputState, {}> {
 readonly state: CheckboxInputState = {
    checked: this.props.checked,
    inputId: this.props.inputId || uuid(),
  };

  handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    this.setState({ checked: e.target.checked });
    this.props.onChange!({[this.props.name]: e.target.checked});
  };

  render() {
    const { checked, inputId } = this.state;
    const { label } = this.props;

    return (
      <Container>
        <Input
          id={inputId}
          checked={checked}
          onChange={this.handleChange}
        />
        <Label htmlFor={inputId} />
        <HiddenLabel>{label}</HiddenLabel>
    </Container>
    );
  }
}

export default CheckboxInput;
