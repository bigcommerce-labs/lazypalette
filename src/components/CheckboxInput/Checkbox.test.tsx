import { shallow } from 'enzyme';
import React from 'react';

import CheckboxInput from './CheckboxInput';

const id = 'CheckboxTestId';

it('renders', () => {
  const checkbox = shallow(
    <CheckboxInput
        checked={true}
        inputId={id}
        label="Label"
        name="show_accept_amex"/>
  );
  expect(checkbox).toMatchSnapshot();
});
