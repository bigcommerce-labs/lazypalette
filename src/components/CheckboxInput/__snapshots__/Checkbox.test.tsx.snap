// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<styled.div>
  <styled.input
    checked={true}
    id="CheckboxTestId"
    onChange={[Function]}
  />
  <styled.label
    htmlFor="CheckboxTestId"
  />
  <styled.label>
    Label
  </styled.label>
</styled.div>
`;
